lib/ansible/module_utils/junos.py

```diff
@@ -44,7 +44,7 @@
 def check_args(module, warnings):
     provider = module.params['provider'] or {}
     for key in junos_argument_spec:
-        if key in ('provider',) and module.params[key]:
+        if key not in ('provider',) and module.params[key]:
```

Comment:
I'll ping @privateip as I believe he wanted to get rid of first level arguments, other than provider.